diff --git a/package/kernel/linux/modules/netfilter.mk b/package/kernel/linux/modules/netfilter.mk
index 0134e81e9e..1ddb24b2fa 100644
--- a/package/kernel/linux/modules/netfilter.mk
+++ b/package/kernel/linux/modules/netfilter.mk
@@ -1256,6 +1256,22 @@ endef
 $(eval $(call KernelPackage,nfnetlink-cttimeout))
 
 
+define KernelPackage/nf-conntrack-bridge
+  SUBMENU:=$(NF_MENU)
+  TITLE:=IPv4/IPv6 bridge connection tracking support
+  DEPENDS:=+kmod-nf-conntrack
+  FILES:=$(LINUX_DIR)/net/bridge/netfilter/nf_conntrack_bridge.ko
+  KCONFIG:=CONFIG_NF_CONNTRACK_BRIDGE
+  AUTOLOAD:=$(call AutoProbe,nf_conntrack_bridge)
+endef
+
+define KernelPackage/nf-conntrack-bridge/description
+ Native bridge connection tracking for nftables flow offload.
+endef
+
+$(eval $(call KernelPackage,nf-conntrack-bridge))
+
+
 define KernelPackage/nf-conntrack-netlink
   TITLE:=Connection tracking netlink interface
   FILES:=$(LINUX_DIR)/net/netfilter/nf_conntrack_netlink.ko
